src/services/userService.js

```diff
@@ -0,0 +1,50 @@
+let users = [];
+let userId = 1;
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllusers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((user) => user.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    name,
+    id: userId,
+  };
+
+  userId++;
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(id);
+
+  Object.assign(user, { name });
```

Comment:
The `updateUser` function does not handle the case where the user with the given ID is not found. Consider adding a check to return `null` or an appropriate response if the user is not found before attempting to update it.